compiler/rustc_infer/src/infer/relate/type_relating.rs

```diff
@@ -13,9 +14,38 @@ use crate::infer::{DefineOpaqueTypes, InferCtxt, SubregionOrigin};
 
 /// Enforce that `a` is equal to or a subtype of `b`.
 pub struct TypeRelating<'combine, 'a, 'tcx> {
+    // Immutable except for the `InferCtxt` and the
+    // resulting nested `goals`.
     fields: &'combine mut CombineFields<'a, 'tcx>,
+
+    // Immutable field.
     structurally_relate_aliases: StructurallyRelateAliases,
+    // Mutable field.
     ambient_variance: ty::Variance,
+
+    /// The cache has only tracks the `ambient_variance` as its the
```

Comment:
(sorry for the piecemeal delivery I forgot to submit this one earlier)

```suggestion
    /// The cache only tracks the `ambient_variance` as it's the
```